### PR TITLE
create recipe for ncdu 1.14.1

### PR DESCRIPTION
**Pull request by Ra-V2**

### CODE REVIEW

File: sys-fs/ncdu/ncdu-1.14.1.recipe [hunk 1]
```diff
@@ -0,0 +1,39 @@
+SUMMARY="A disk usage analyser with a ncurses interface"
+DESCRIPTION="Ncdu aims to be fast, simple and easy to use, and should be able to run in any minimal POSIX-like environment with ncurses installed."
```
Comment (extrowerk) [inline sys-fs/ncdu/ncdu-1.14.1.recipe]: Please break the lines at 80 char.

File: sys-fs/ncdu/ncdu-1.14.1.recipe [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+HOMEPAGE="https://dev.yorhel.nl/ncdu"
+LICENSE="MIT"
+COPYRIGHT="Yoran Heling"
+SOURCE_URI="https://dev.yorhel.nl/download/ncdu-$portVersion.tar.gz"
+CHECKSUM_SHA256="be31e0e8c13a0189f2a186936f7e298c6390ebdc573bb4a1330bc1fcbf56e13e"
+SOURCE_DIR="ncdu-$portVersion"
```
Comment (extrowerk) [inline sys-fs/ncdu/ncdu-1.14.1.recipe]: I think this is the default, so it is unnecesary.

File: sys-fs/ncdu/ncdu-1.14.1.recipe [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+REVISION="1"
```
Comment (extrowerk) [inline sys-fs/ncdu/ncdu-1.14.1.recipe]: The ordering is wrong, please check the wiki.

File: sys-fs/ncdu/ncdu-1.14.1.recipe [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+	"
+BUILD_PREREQUIRES="
+	cmd:gcc
+	cmd:make
+	cmd:autoconf
+	"
+
+BUILD()
+{
+	runConfigure ./configure --prefix=/$prefix
```
Comment (extrowerk) [inline sys-fs/ncdu/ncdu-1.14.1.recipe]: The / dont required for $prefix

File: sys-fs/ncdu/ncdu-1.14.1.recipe [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+	lib:libncurses
+	"
+BUILD_REQUIRES="
+	haiku_devel
+	devel:libncurses
+	"
+BUILD_PREREQUIRES="
+	cmd:gcc
+	cmd:make
+	cmd:autoconf
```
Comment (return) [inline sys-fs/ncdu/ncdu-1.14.1.recipe]: The BUILD_PREREQUIRES should also be in alphabetical order.

```suggestion
	cmd:autoconf
	cmd:gcc
	cmd:make
```

File: sys-fs/ncdu/ncdu-1.14.1.recipe [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+and should be able to run in any minimal POSIX-like environment \
+with ncurses installed."
+HOMEPAGE="https://dev.yorhel.nl/ncdu"
+COPYRIGHT="Yoran Heling"
+LICENSE="MIT"
+REVISION="1"
+SOURCE_URI="https://dev.yorhel.nl/download/ncdu-$portVersion.tar.gz"
+CHECKSUM_SHA256="be31e0e8c13a0189f2a186936f7e298c6390ebdc573bb4a1330bc1fcbf56e13e"
+
+ARCHITECTURES="!x86_gcc2 !x86 x86_64"
```
Comment (Begasus) [inline sys-fs/ncdu/ncdu-1.14.1.recipe]: Doesn't it build?

File: sys-fs/ncdu/ncdu-1.14.1.recipe [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+
+PROVIDES="
```
Comment (Begasus) [inline sys-fs/ncdu/ncdu-1.14.1.recipe]: Needs something to provide:
```diff
+	ncdu = $portVersion
```

File: sys-fs/ncdu/ncdu-1.14.1.recipe [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+	"
+BUILD_PREREQUIRES="
+	cmd:autoconf
+	cmd:gcc
+	cmd:make
+	"
+
+BUILD()
+{
+	runConfigure ./configure --prefix=$prefix
```
Comment (Begasus) [inline sys-fs/ncdu/ncdu-1.14.1.recipe]: No need for prefix runConfigure taks care of that:
```diff
-	runConfigure ./configure --prefix=$prefix
+	runConfigure ./configure
´´´

File: sys-fs/ncdu/ncdu-1.14.1.recipe [hunk 9]
```diff
@@ -0,0 +1,40 @@
+SUMMARY="A disk usage analyser with a ncurses interface"
+DESCRIPTION="Ncdu aims to be fast, simple and easy to use, \
+and should be able to run in any minimal POSIX-like environment \
+with ncurses installed."
+HOMEPAGE="https://dev.yorhel.nl/ncdu"
+COPYRIGHT="Yoran Heling"
```
Comment (Begasus) [inline sys-fs/ncdu/ncdu-1.14.1.recipe]: Add year(s) to it:
```diff
-COPYRIGHT="Yoran Heling"
+COPYRIGHT="2007-2019 Yoran Heling"
```

**Comment by Begasus** [comment]
This can be build for secondary architecture on gcc2hybrid, could you look into that?
ncdu

**Comment by Ra-V2** [comment]
I have no idea what I should do next to make it work. Any tips? Maybe I should patch the source or something? When i try to build i get many errors in code.

**Comment by alaviss** [comment]
Try adding `cmd:libtoolize$secondaryArchSuffix`, then call `libtoolize -fci" before you run `configure`.

File: sys-fs/ncdu/ncdu-1.14.1.recipe [hunk 10]
```diff
@@ -0,0 +1,45 @@
...
+REVISION="1"
+SOURCE_URI="https://dev.yorhel.nl/download/ncdu-$portVersion.tar.gz"
+CHECKSUM_SHA256="be31e0e8c13a0189f2a186936f7e298c6390ebdc573bb4a1330bc1fcbf56e13e"
+
+ARCHITECTURES="x86_gcc2 x86 x86_64"
+SECONDARY_ARCHITECTURES="x86"
+
+PROVIDES="
+	ncdu = $portVersion
+	cmd:ncdu = $portVersion
```
Comment (Begasus) [inline sys-fs/ncdu/ncdu-1.14.1.recipe]: Add secondaryArchSuffix here too:
```diff
-	ncdu = $portVersion
-	cmd:ncdu = $portVersion
+	ncdu$secondaryArchSuffix = $portVersion
+	cmd:ncdu$secondaryArchSuffix = $portVersion
```
You could pass commandBindDir here (option)

File: sys-fs/ncdu/ncdu-1.14.1.recipe [hunk 11]
```diff
@@ -0,0 +1,45 @@
...
+REQUIRES="
+	haiku$secondaryArchSuffix
+	lib:libncurses$secondaryArchSuffix
+	"
+BUILD_REQUIRES="
+	haiku${secondaryArchSuffix}_devel
+	devel:libncurses$secondaryArchSuffix
+	"
+BUILD_PREREQUIRES="
+	cmd:autoconf$secondaryArchSuffix
```
Comment (Begasus) [inline sys-fs/ncdu/ncdu-1.14.1.recipe]: Need to check on x86_64, on _x86 it doesn't need autoconf, but it needs "cmd:awk"

File: sys-fs/ncdu/ncdu-1.14.1.recipe [hunk 12]
```diff
@@ -0,0 +1,45 @@
...
+	cmd:gcc$secondaryArchSuffix
+	cmd:libtool$secondaryArchSuffix
```
Comment (Begasus) [inline sys-fs/ncdu/ncdu-1.14.1.recipe]: libtool not needed can be removed

File: sys-fs/ncdu/ncdu-1.14.1.recipe [hunk 13]
```diff
@@ -0,0 +1,45 @@
...
+	cmd:make
+	cmd:perl
```
Comment (Begasus) [inline sys-fs/ncdu/ncdu-1.14.1.recipe]: perl not needed, can be removed (check also for 64-bit)

File: sys-fs/ncdu/ncdu-1.14.1.recipe [hunk 14]
```diff
@@ -0,0 +1,45 @@
...
+	cmd:pkg_config$secondaryArchSuffix
```
Comment (Begasus) [inline sys-fs/ncdu/ncdu-1.14.1.recipe]: works ok without here, but shouldn't harm to leave it in

**Comment by Begasus** [comment]
Shouldn't need to run libtoolize @alaviss 
@Ra-V2 what command did you gave in Terminal? 
"haikuporter ncdu" or "haikuporter ncdu_x86" (last one is correct)

**Comment by Ra-V2** [comment]
I didn't have secondary architecture set in the configuration file :)

**Comment by Ra-V2** [comment]
screenshot1 (1) (1)

File: sys-fs/ncdu/ncdu-1.14.1.recipe [hunk 15]
```diff
@@ -15,8 +15,8 @@ SECONDARY_ARCHITECTURES="x86"
 PROVIDES="
 	ncdu$secondaryArchSuffix = $portVersion
 	cmd:ncdu$secondaryArchSuffix = $portVersion
- 	commandBinDir = $binDir
 	"
+
```
Comment (return) [inline sys-fs/ncdu/ncdu-1.14.1.recipe]: There shouldn't be a space between PROVIDES and REQUIRES. Please remove the extra space here on line 19.

File: sys-fs/ncdu/ncdu-1.14.1.recipe [hunk 16]
```diff
@@ -15,8 +15,8 @@ SECONDARY_ARCHITECTURES="x86"
...
 REQUIRES="
 	haiku$secondaryArchSuffix
 	lib:libncurses$secondaryArchSuffix
```
Comment (return) [inline sys-fs/ncdu/ncdu-1.14.1.recipe]: Please add a single space on line 23. There should be a space between REQUIRES and BUILD_REQUIRES.

File: sys-fs/ncdu/ncdu-1.14.1.recipe (hunk 15 above)
Comment (return) [reply sys-fs/ncdu/ncdu-1.14.1.recipe]: @Ra-V2 It seems you have applied this change before I reviewed your previous changes. You can ignore this suggestion as it is outdated.

**Comment by Begasus** [comment]
Thanks for the contribution @Ra-V2 :+1:

**Pull request merged**